hierarqcal/core.py

```diff
@@ -636,23 +650,188 @@ def __eq__(self, other):
+
+        Args:
+            Qc_l (list): List of available qubits.
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments, such as:
+                * mapping (tuple(function, int)): TODO update this docstring, it's not a tuple anymore.
+                    Function mapping is specified as a tuple, where the first argument is a function and the second is the number of symbols it uses. A symbol here refers to an variational paramater for a quantum circuit, i.e. crz(theta, q0, q1) <- theta is a symbol for the gate.
+
+        Returns:
+            Qconv: Returns the updated version of itself, with correct nodes and edges.
+        """
+
+        self.pivot_pattern_fn = self.get_pivot_pattern_fn(self.pattern, Qc_l)
+        pivot_q = self.pivot_pattern_fn(Qc_l)
+        remaining_q = [q for q in Qc_l if q not in pivot_q]
+
+        if self.arity > 1:
+            nq_available = len(remaining_q)
+            if nq_available > 0:
+                if self.stride % nq_available == 0:
+                    # TODO make this clear in documentation
+                    # warnings.warn(
+                    #     f"Stride and number of available qubits can't be the same, received:\nstride: {self.stride}\n available qubits:{nq_available}. Defaulting to stride of 1"
+                    # )
+                    self.stride = 1
+                if self.boundary == "open":
+                    mod_nq = lambda x: x % nq_available
+                    Ec_l = [
+                        tuple(
+                            (
+                                remaining_q[i + j * self.stride]
+                                for j in range(self.arity-1)
+                                if i + j * self.stride < nq_available
+                            )
+                        )
+                        for i in range(self.offset, nq_available, self.step)
+                    ]
+                    
+
+                else:
+                    mod_nq = lambda x: x % nq_available
+                    Ec_l = [
+                        tuple(
+                            (Qc_l[mod_nq(i + j * self.stride)] for j in range(self.arity-1))
+                        )
+                        for i in range(self.offset, nq_available, self.step)
+                    ]
+
+                # Add the pivot qubit to each edge
+                Ec_l = Ec_l[: len(Ec_l) - len(Ec_l) % len(pivot_q)]
+                Ec_l = [
+                    tuple(edge+(p,))
+                    for p, edge in zip(
+                        [P for P in pivot_q for _ in range(len(Ec_l) // len(pivot_q))],
+                        Ec_l,
+                    )
+                ]
+                # Remove all that is not "complete"
+                Ec_l = [edge for edge in Ec_l if len(edge) == self.arity]
```

Comment:
I think another "clean up" step is required consider the edges of this case:
```python
N=10
u = Qunitary("cnot()^10")
qft = Qinit(N)  + Qpivot("1*1*1", stride=2,step=2, mapping=u, boundary="periodic")
circuit= qft(backend="qiskit")
circuit.draw("mpl")
```
This breaks because qft[1].E is: `[('q1', 'q1'), ('q3', 'q6'), ('q5', 'q10')]`. So a fix can be to just remove any edge that contains duplicate qubits. 

It's not entirely clear what the above pivot should be, but having this check in general would be useful. I have done this for mask before, so you can use:
```python
Ep_l = [edge for edge in Ep_l if len(set(edge)) == self.arity]
```